### PR TITLE
Installer: encrypt all non-root file-systems.

### PR DESCRIPTION
**Pull request by rgooch**
The installer will now encrypt all non-root file-systems, storing an access key on the root file-system.
Performance of the installer has been improved by making the operations heavily concurrent. For a system with as few as 4 drives, this improves installation time by over one minute.

### CODE REVIEW

File: cmd/installer/configureStorage.go [hunk 1]
```diff
@@ -262,15 +375,36 @@ func getImage(logger log.DebugLogger) (*image.Image, *srpc.Client, error) {
...
 		return nil, nil, err
 	} else {
-		logger.Debugf(0, "got image: %s\n", imageName)
+		logger.Debugf(0, "got image: %s in %s\n",
+			imageName, format.Duration(time.Since(startTime)))
 		return img, client, nil
 	}
 }
 
+func getRandomKey(numBytes uint, logger log.DebugLogger) ([]byte, error) {
```
Comment (cviecco) [inline cmd/installer/configureStorage.go]: use cryto/random instead of this

File: cmd/installer/configureStorage.go [hunk 2]
```diff
@@ -52,8 +59,94 @@ func init() {
...
+		sizeInUnits := partition.MinimumFreeBytes / unitSize
+		if sizeInUnits*unitSize < partition.MinimumFreeBytes {
+			sizeInUnits++
+		}
+		args = append(args, "mkpart", "primary", "ext2",
+			strconv.FormatUint(offsetInUnits, 10)+unitSuffix,
+			strconv.FormatUint(offsetInUnits+sizeInUnits, 10)+unitSuffix)
+		offsetInUnits += sizeInUnits
+	}
+	args = append(args, "mkpart", "primary", "ext2",
```
Comment (cviecco) [inline cmd/installer/configureStorage.go]: why ext2

File: cmd/installer/configureStorage.go [hunk 3]
```diff
@@ -52,8 +59,94 @@ func init() {
...
+		strconv.FormatUint(offsetInUnits, 10)+unitSuffix, "100%")
+	args = append(args,
+		"set", strconv.FormatInt(int64(bootPartition), 10), "boot", "on")
+	if err := run("parted", *tmpRoot, logger, args...); err != nil {
+		return err
+	}
+	// Prepare all file-systems concurrently, make them serially.
+	concurrentState := concurrent.NewState(uint(
```
Comment (cviecco) [inline cmd/installer/configureStorage.go]: Concurrent? this is new lib? what does it get you?

File: cmd/installer/configureStorage.go (hunk 1 above)
Comment (rgooch) [reply cmd/installer/configureStorage.go]: I already have a change cooking for this, but I'm testing some odd performance behaviour. I'd like to defer this for now.

File: cmd/installer/configureStorage.go (hunk 2 above)
Comment (rgooch) [reply cmd/installer/configureStorage.go]: That's the partition type for ext{2,3,4}.

File: cmd/installer/configureStorage.go (hunk 3 above)
Comment (rgooch) [reply cmd/installer/configureStorage.go]: It makes it easier to run goroutines and reap (and stop) the error code. It's been around since Feb-2016.

**Pull request merged**